doc/gin_files/version_converter.py

```diff
@@ -0,0 +1,212 @@
+                            if val_elem.tag != "value" and one_value:
+                                elem_name = VersionConverter._version_map[val_elem.tag] \
+                                    if val_elem.tag in VersionConverter._version_map else val_elem.tag
+                                new_elem = ET.Element(elem_name)
+                                new_elem.text = val_elem.text
+                                value.getparent().append(new_elem)  # appending to the property
+                                value.remove(val_elem) 
+                        one_value = False
+                    else:
+                        first_value.text += ", " + value.text
+                        prop.remove(value)
+        return tree
+
+    @staticmethod
+    def validate_xml(filename):
+        """
+        Fix an xml file by deleting known mismatching tags 
+        """
+        changes_list = []
+        with open(filename) as f:
+            doc = f.read()
+            for k, v in VersionConverter._error_strings.items():
+                if k not in doc:
+                    continue
+                else:
+                    changes_list.append(k)
+
+        if len(changes_list) == 0:
+            return
+
+        # Write the changed content, if found in the file
+        with open(filename, 'w') as f:
+            for str in changes_list:
+                doc = doc.replace(str, VersionConverter._error_strings[str])
+            f.write(doc)
+
+    @staticmethod
+    def _replace_same_name_entities(tree):
+        """
+        Changes same section names in the doc by adding <-{index}> to the next section occurrences
+        :param tree: ElementTree of the doc
+        :return: ElementTree
+        """
+        section_map = {}
+        root = tree.getroot()
+        for section in root.iter("section"):
+            n = section.find("name")
+            if n.text not in section_map:
+                section_map[n.text] = 1
+            else:
+                section_map[n.text] += 1
+                n.text += "-" + str(section_map[n.text])
+        return tree
+
+    def __str__(self):
+        tree = self.convert_odml_file(self.filename)
+        return ET.tounicode(tree, pretty_print=True) if tree else ""
+
+    def __unicode__(self):
+        tree = self.convert_odml_file(self.filename)
+        return ET.tounicode(tree, pretty_print=True) if tree else ""
+    
+    def write_to_file(self, filename):
+        if sys.version_info < (3,):
+            data = unicode(self).encode('utf-8')
+        else:
+            data = str(self)
+        if data and "<odML " in data:
+            f = open(filename, "w")
+            f.write(self.header)
+            f.write(data)
+            f.close()
+        else:
+            print("Empty file: ", filename)
+
+
+class DirectoryParser:
+    def __init__(self):
+        self.doc = None
+
+    def convert_directory(self, directory_path, suffix, extension):
```

Comment:
The usage of `suffix` is a bit unclear to me. If I interpret correctly: With it, the user is given the choice of how the folder containing the result files should be suffixed with. But only if the suffix is literally `_updated` or `_odmls`, then these folders will be excluded from a second round of conversion, if it is run again. If the suffix is any other string, a second conversion will include these result files as well. Maybe we should either allow only `_updated` or `_odmls` as valid suffix strings or explicitly tell the user in the docstring, what can happen if a different suffix is used.